content/sources/contributions.adoc

```diff
@@ -0,0 +1,374 @@
+= Contributions
+
+== Workflow
+
+The contribution workflow mainly relies on Github Pull Requests. Here are the rules :
+
+* `PR for everything` +
+For every fix/feature, for all platform projects and supported addons, create a PR.
+Creating a PR is easy and allows to share, discuss on validate more easily the fix/feature.
+To create a PR :
+** create a branch locally with your fix and push it to Github
+** in Github, select your branch, and click on *New pull request* image:git-pr-create-01.png[alt="Git PR Create",align="center",width="700"]
+** select the right base branch (most of the time *develop*) and check that the PR contains only your commits image:git-pr-create-02.png[alt="Git PR Create",width="700",align="center"]
+** fill the description
+** click on *Create pull request*
+* `PR on develop` +
+++
+PR must always be done on exodev:develop or exo-addons:develop, not on exoplatform:develop (we fix on develop first, then we backport to stable if needed).
+Remember that exoplatform:develop is read-only.
+There are only 2 exceptions :
+++
+** the fix is different between develop dans stable - in such a case, 2 PRs are necessary, one on develop, one on stable
+** the bug only occurs on stable - in such a case, PR must be done on stable
+* `Explain your PR`
++
+The description of the PR must always be filled to describe precisely what was the root cause of the problem and how it has been solved.
+* `Update the PR, do not recreate it`
++
+To update a PR, just push a new commit in the same branch, no need to create a new branch and a new PR.
+Creating a new PR for each update prevents from following easily the discussion and the updates history.
+* `PR must be validated by peers`
++
+When a PR is submitted, it has to be reviewed by at least one peer.
+When the PR is validated by the peer, the PR can be merged in the target branch.
+* `Merge the PR correctly`
++
+Before merging the PR in the target branch, make sure the branch of the PR is up to date (push --force), otherwise the PR will not appear as *Merged* in Github.
```

Comment:
why do you mention `(push --force)` to check if the PR branch is up to date ? Should not you talk about `rebase` instead ?

Reply:
The remark is about making sure the remote branch is aligned with the local branch before merging (according to the process described in http://developer.exoplatform.org/#id-fix-branch), otherwise the PR will not be marked as Merged. But I agree it would deserve a bit more explanations and rework (we should also talk about merging through the UI for example).
But actually I did not update this part in this PR, I just moved it.

Reply:
If the target branch security is well configured, force push local changes to the remote branch after the PR validation will invalidate the PR ...

Reply:
Yes